### PR TITLE
Completed system sign-in

### PR DESCRIPTION
**Pull request by 2AStudios**

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 1]
```diff
@@ -42,9 +54,10 @@
      */
     private ViewPager mViewPager;
 
-
     final int LOGIN_REQ_CODE = 10031;
 
+    UserSession userSession;
```
Comment (JehronPett) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: Make this a class field of `RetrieveUserSessionTask`

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 2]
```diff
@@ -109,23 +122,50 @@ public void onClick(View v) {
 //            }
 //        });
 
-        Intent signInIntent = new Intent(this, LoginActivity.class);
-        startActivityForResult(signInIntent, LOGIN_REQ_CODE);
-
+        //Force sign out every app launch, for debugging purposes only
```
Comment (JehronPett) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: Do we need this anymore?

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 3]
```diff
@@ -109,23 +122,50 @@ public void onClick(View v) {
...
     }
 
     @Override
-    public void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if (requestCode == LOGIN_REQ_CODE) {
-            if (resultCode == RESULT_OK) {
+    protected void onStart() {
+        super.onStart();
+        GoogleSignInAccount account = GoogleSignIn.getLastSignedInAccount(this);
+        //        //If account is null, attempt to sign in, if not, launch the normal activity. updateUI(account);
```
Comment (JehronPett) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]:
```suggestion
        // If account is null, attempt to sign in, if not, launch the normal activity. updateUI(account);
```

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 4]
```diff
@@ -109,23 +122,50 @@ public void onClick(View v) {
...
+        if (account == null) {
+            Intent signInIntent = new Intent(this, LoginActivity.class);
```
Comment (JehronPett) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]:
```suggestion
            final Intent signInIntent = new Intent(this, LoginActivity.class);
```

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 5]
```diff
@@ -109,23 +122,50 @@ public void onClick(View v) {
...
+        GoogleSignIn.getClient(this, gso).signOut();
     }
 
     @Override
-    public void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if (requestCode == LOGIN_REQ_CODE) {
-            if (resultCode == RESULT_OK) {
+    protected void onStart() {
+        super.onStart();
+        GoogleSignInAccount account = GoogleSignIn.getLastSignedInAccount(this);
```
Comment (JehronPett) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]:
```suggestion
        final GoogleSignInAccount account = GoogleSignIn.getLastSignedInAccount(this);
```

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 6]
```diff
@@ -42,9 +54,10 @@
      */
     private ViewPager mViewPager;
 
-
     final int LOGIN_REQ_CODE = 10031;
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: can't this be `private static final`?

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java (hunk 2 above)
Comment (young-k) [reply app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: There should also be a space after `//`

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 7]
```diff
@@ -109,23 +122,50 @@ public void onClick(View v) {
...
+                        account.getId(), account.getGivenName(),
+                        account.getFamilyName(), account.getEmail()));
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return userSession;
+        }
+
+        protected void onPostExecute(UserSession userSession) {
+            if (userSession != null) {
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: nitpicky, but:

```
if (userSession == null) return;
/* rest of code here */
```

This reduces the level of indentation

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java [hunk 8]
```diff
@@ -19,20 +20,18 @@
 public final class NetworkUtils {
 
     private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
-    private static final String MOBILE_AUTH_ROUTE = "/auth/mobile";
+    private static final String MOBILE_AUTH_ROUTE = "/api/v2/auth/mobile";
+    static final String DEPLOYED_BACKEND = "http://pollo-backend.cornellappdev.com";
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: why not `private`? also this variable isn't properly named, can we change it to something like `BASE_URL` or something?

File: app/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -9,21 +9,21 @@
...
             </intent-filter>
         </activity>
-        <activity android:name=".LoginActivity"
-            android:theme="@style/AppTheme.NoActionBar">
-
-        </activity>
     </application>
 
+    <uses-permission android:name="android.permission.INTERNET"/>
+
```
Comment (young-k) [inline app/src/main/AndroidManifest.xml]: why is there a `\n` here? is this customary? 

cc: @ZiweiGu @JehronPett @abdullah248 @ningning621

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java (hunk 2 above)
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: > Do we need this anymore?

We don't particularly need it anymore, but I'm not removing the text entirely until we implement a working logout button Until then, this is our best way of interacting with the login system.

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java (hunk 1 above)
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: I don't entirely understand what you mean by that, but we can meet about this later.

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 10]
```diff
@@ -109,23 +122,50 @@ public void onClick(View v) {
...
+                e.printStackTrace();
             }
+            return userSession;
+        }
+
+        protected void onPostExecute(UserSession userSession) {
+            if(userSession == null) return;
+
+            String loginText = "Logged into ";
+            loginText += userSession.getAccessToken();
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: why not just put this on the previous line?

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 11]
```diff
@@ -109,23 +122,50 @@ public void onClick(View v) {
...
+                        account.getId(), account.getGivenName(),
+                        account.getFamilyName(), account.getEmail()));
+            } catch (IOException e) {
+                e.printStackTrace();
             }
+            return userSession;
+        }
+
+        protected void onPostExecute(UserSession userSession) {
+            if(userSession == null) return;
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: space after `if`

**Pull request merged**